src/mlpack/core/tree/cover_tree/cover_tree_impl.hpp

```diff
@@ -547,6 +548,72 @@ CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::CoverTree(
   }
 }
 
+// Copy Assignment.
+template<
+    typename MetricType,
+    typename StatisticType,
+    typename MatType,
+    typename RootPointPolicy
+>
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>&
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::
+operator=(const CoverTree& other)
+{
+  if (this == &other)
+    return *this;
+
+  // Freeing memory that will not be used anymore.
+  delete dataset;
```

Comment:
Take a look at the destructor of `CoverTree` to see what members need to be deleted (and how to do it).  We also need to consider `metric` here. :+1: